diff --git b/polybar/.config/polybar/polybar-scripts/info-hackspeed.sh a/polybar/.config/polybar/polybar-scripts/info-hackspeed.sh
index 999c31e..5bf78af 100755
--- b/polybar/.config/polybar/polybar-scripts/info-hackspeed.sh
+++ a/polybar/.config/polybar/polybar-scripts/info-hackspeed.sh
@@ -1,12 +1,13 @@
 #!/bin/sh
 # shellcheck disable=SC2016,SC2059
 
-KEYBOARD_ID="AT Translated Set 2 keyboard"
+KEYBOARD_NAME="Logitech Wireless Keyboard PID:4023"
+KEYBOARD_ID=$(xinput list | grep "$KEYBOARD_NAME" | cut -d '=' -f 2 | awk '{ print $1 }' | tail -1)
 
 # cpm: characters per minute
 # wpm: words per minute (1 word = 5 characters)
 METRIC=cpm
-FORMAT="# %d $METRIC"
+FORMAT="%d $METRIC"
 
 INTERVAL=20
 
@@ -20,8 +21,7 @@ case "$LAYOUT" in
 	qwerty) CONDITION='($3 >= 10 && $3 <= 19) || ($3 >= 24 && $3 <= 33) || ($3 >= 37 && $3 <= 53) || ($3 >= 52 && $3 <= 58)'; ;;
 	azerty) CONDITION='($3 >= 10 && $3 <= 19) || ($3 >= 24 && $3 <= 33) || ($3 >= 37 && $3 <= 54) || ($3 >= 52 && $3 <= 57)'; ;;
 	qwertz) CONDITION='($3 >= 10 && $3 <= 20) || ($3 >= 24 && $3 <= 34) || ($3 == 36) || ($3 >= 38 && $3 <= 48) || ($3 >= 52 && $3 <= 58)'; ;;
-    dvorak) CONDITION='($3 >= 10 && $3 <= 19) || ($3 >= 27 && $3 <= 33) || ($3 >= 38 && $3 <= 47) || ($3 >= 53 && $3 <= 61)'; ;;
-    dontcare) CONDITION='1'; ;; # Just register all key presses, not only letters and numbers
+        dontcare) CONDITION='1'; ;; # Just register all key presses, not only letters and numbers
 	*) echo "Unsupported layout \"$LAYOUT\""; exit 1; ;;
 esac
 
diff --git b/polybar/.config/polybar/polybar-scripts/info-softwarecounter.py a/polybar/.config/polybar/polybar-scripts/info-softwarecounter.py
index ad90b7d..13eb708 100755
--- b/polybar/.config/polybar/polybar-scripts/info-softwarecounter.py
+++ a/polybar/.config/polybar/polybar-scripts/info-softwarecounter.py
@@ -1,7 +1,6 @@
 #!/usr/bin/env python3
 
 import subprocess
-import sys
 from collections import Counter, OrderedDict
 
 
@@ -14,36 +13,36 @@ PROCESS = True
 
 # program/process names and corresponding gylphs
 guis = OrderedDict({
-    'terminals': '#',
-    'chromes': '#',
-    'firefox': '#',
-    'skypeforlinux': '#',
-    'filemanager': '#',
-    'remote-desktop': '#',
-    'pdfviewer': '#',
-    'image': '#',
+    'terminals': '%{T10}%{T}',
+    'chromes': '%{T10}%{T}',
+    'firefox': '%{T10}%{T}',
+    'skypeforlinux': '%{T10}瑩%{T}',
+    'filemanager': '%{T10}%{T}',
+    'remote-desktop': '%{T10}ﲾ%{T}',
+    'pdfviewer': '%{T10}%{T}',
+    'image': '%{T10}%{T}',
 })
 
 processes = OrderedDict({
-    'vims': '#',
-    'ssh': '#',
-    'updater': '#',
+    'vims': '%{T10}%{T}',
+    'ssh': '%{T10}%{T}',
+    'updater': '%{T10}%{T}',
 })
 
 # combine counts of program/process names in the tuple
 # the resulting glpyh used will be that of the corresponding key
 combine_guis = {
-    'terminals': ('termite', 'terminator', 'urxvt'),
+    'terminals': ('gnome-terminal-server', 'xfce4-terminal', 'termite', 'terminator', 'urxvt'),
     'chromes': ('chromium', 'chrome'),
-    'filemanger': ('nemo', 'thunar', 'dolphin', 'nautilus', 'pcmanfm'),
+    'filemanger': ('nemo', 'thunar', 'dolphin', 'nautilus', 'pcmanfm', 'Archivos'),
     'remote-desktop': ('TeamViewer', ),
     'pdfviewer': ('evince', 'okular', 'zathura'),
     'image': ('gthumb', 'shotwell', 'deepin-image-vi'),
 }
 
 combine_proccesses = {
-    'vims': ('nvim', 'vim'),
-    'updater': ('pacman', 'yay', 'trizen', 'yaourt'),
+    'vims': ('nvim', 'vim', 'atom'),
+    'updater': ('pacman', 'yay', 'trizen', 'yaourt', 'makepkg', 'auracle'),
 }
 
 
@@ -53,12 +52,9 @@ process_output = ''
 if GUI:
 
     def get_running_guis():
-        try:
-            listed = sys.argv[1]
-        except IndexError:
-            listed = []
 
-        get = lambda cmd: subprocess.check_output(cmd).decode("utf-8").strip()
+        def get(cmd):
+            return subprocess.check_output(cmd).decode("utf-8").strip()
 
         def check_wtype(w_id):
             # check the type of window, only list "NORMAL" windows
@@ -66,19 +62,30 @@ if GUI:
 
         def get_process(w_id):
             # get the name of the process, owning the window
-            proc = get(["ps", "-p", w_id, "-o", "comm="])
-            return proc
+            proc = get(["ps", "-p", w_id, "-o", "cmd="])
+            return proc.split()[0].split("/")[-1]
+
+        wlist = [
+            line.split() for line in subprocess.check_output(["wmctrl", "-lp"]).decode("utf-8").splitlines()
+        ]
+
+        # Most valuable deubg info! #
+        # print("wlist after LC:")
+        # print("---------")
+        # for i in wlist:
+        #     print(i)
+        # print("---------")
+        # ######################### #
 
-        wlist = [l.split() for l in subprocess.check_output(["wmctrl", "-lp"])\
-                 .decode("utf-8").splitlines()]
         validprocs = [
-            get_process(w[2]) for w in wlist if check_wtype(w[0]) == True and w[2] != '0'
+            get_process(w[2]) for w in wlist if check_wtype(w[0]) and w[2] != '0'
         ]
 
         return validprocs
 
     # get list of running GUI programs
     gui_counts = Counter(get_running_guis())
+    # print(gui_counts)  # Another interesting debug statement
 
     # combine programs in program combine list
     for k, lst in combine_guis.items():
@@ -97,7 +104,7 @@ if GUI:
             c = gui_counts[k]
             if c:
                 gui_output += '%s %i ' % (v, c)
-        except:
+        except Exception:
             pass
 
 if PROCESS:
@@ -108,12 +115,14 @@ if PROCESS:
         for i, p in enumerate(process_name_list):
             try:
                 count = int(
-                    subprocess.check_output(['pgrep', '-c', '-x',
-                                             p]).decode('utf-8'))
+                    subprocess.check_output(
+                        ['pgrep', '-c', '-x', p]
+                    ).decode('utf-8')
+                )
             except subprocess.CalledProcessError:
                 count = 0
-            counts[i] = (p, count)
 
+            counts[i] = (p, count)
         return dict(counts)
 
     # count running proccesses
@@ -139,7 +148,7 @@ if PROCESS:
             c = process_counts[k]
             if c:
                 process_output += '%s %i ' % (v, c)
-        except:
+        except Exception:
             pass
 
 print(gui_output + process_output)
diff --git b/polybar/.config/polybar/polybar-scripts/openweathermap-fullfeatured.sh a/polybar/.config/polybar/polybar-scripts/openweathermap-fullfeatured.sh
index 652c620..d5f92f4 100755
--- b/polybar/.config/polybar/polybar-scripts/openweathermap-fullfeatured.sh
+++ a/polybar/.config/polybar/polybar-scripts/openweathermap-fullfeatured.sh
@@ -3,23 +3,23 @@
 get_icon() {
     case $1 in
         # Icons for weather-icons
-        01d) icon="";;
-        01n) icon="";;
-        02d) icon="";;
-        02n) icon="";;
-        03*) icon="";;
-        04*) icon="";;
-        09d) icon="";;
-        09n) icon="";;
-        10d) icon="";;
-        10n) icon="";;
-        11d) icon="";;
-        11n) icon="";;
-        13d) icon="";;
-        13n) icon="";;
-        50d) icon="";;
-        50n) icon="";;
-        *) icon="";
+        01d) icon="%{T8} %{T}";;
+        01n) icon="%{T8} %{T}";;
+        02d) icon="%{T8} %{T}";;
+        02n) icon="%{T8} %{T}";;
+        03*) icon="%{T8} %{T}";;
+        04*) icon="%{T8} %{T}";;
+        09d) icon="%{T8} %{T}";;
+        09n) icon="%{T8} %{T}";;
+        10d) icon="%{T8} %{T}";;
+        10n) icon="%{T8} %{T}";;
+        11d) icon="%{T8} %{T}";;
+        11n) icon="%{T8} %{T}";;
+        13d) icon="%{T8} %{T}";;
+        13n) icon="%{T8} %{T}";;
+        50d) icon="%{T8} %{T}";;
+        50n) icon="%{T8} %{T}";;
+        *) icon="%{T8} %{T}";
 
         # Icons for Font Awesome 5 Pro
         #01d) icon="";;
@@ -38,7 +38,7 @@ get_icon() {
         #*) icon="";
     esac
 
-    echo $icon
+    echo "$icon"
 }
 
 get_duration() {
@@ -52,8 +52,8 @@ get_duration() {
 
 }
 
-KEY=""
-CITY=""
+KEY="a94a5e59327331833dd5e93d9b64885c"
+CITY="Getafe,ES"
 UNITS="metric"
 SYMBOL="°"
 
@@ -89,11 +89,11 @@ if [ -n "$current" ] && [ -n "$forecast" ]; then
 
 
     if [ "$current_temp" -gt "$forecast_temp" ]; then
-        trend=""
+        trend="%{T8}%{T}"
     elif [ "$forecast_temp" -gt "$current_temp" ]; then
-        trend=""
+        trend="%{T8} %{T}"
     else
-        trend=""
+        trend="%{T8}%{T}"
     fi
 
 
@@ -102,11 +102,11 @@ if [ -n "$current" ] && [ -n "$forecast" ]; then
     now=$(date +%s)
 
     if [ "$sun_rise" -gt "$now" ]; then
-        daytime=" $(get_duration "$((sun_rise-now))")"
+        daytime="%{T8}%{T} $(get_duration "$((sun_rise-now))")"
     elif [ "$sun_set" -gt "$now" ]; then
-        daytime=" $(get_duration "$((sun_set-now))")"
+        daytime="%{T8}%{T} $(get_duration "$((sun_set-now))")"
     else
-        daytime=" $(get_duration "$((sun_rise-now))")"
+        daytime="%{T8}%{T} $(get_duration "$((sun_rise-now))")"
     fi
 
     echo "$(get_icon "$current_icon") $current_temp$SYMBOL  $trend  $(get_icon "$forecast_icon") $forecast_temp$SYMBOL   $daytime"
diff --git b/polybar/.config/polybar/polybar-scripts/system-cpu-temppercore.sh a/polybar/.config/polybar/polybar-scripts/system-cpu-temppercore.sh
index 1ebc3b2..eb96d0b 100755
--- b/polybar/.config/polybar/polybar-scripts/system-cpu-temppercore.sh
+++ a/polybar/.config/polybar/polybar-scripts/system-cpu-temppercore.sh
@@ -1,3 +1,8 @@
 #!/bin/sh
 
-sensors | grep Core | awk '{print substr($3, 2, length($3)-5)}' | tr "\\n" " " | sed 's/ /°C  /g' | sed 's/  $//'
+#sensors | grep Core | awk '{print substr($3, 2, length($3)-5)}' | tr "\\n" " " | sed 's/ /°C  /g' | sed 's/  $//'
+
+# This is a more accurate way attending to my `sensors` output,
+# though I don't think these the core temperatures, I would say
+# these are another chipsets temps; but at least is something.
+sensors | grep "temp[0-9]" | awk '{print substr($2, 2, length($2)-3)}' | tr "\\n" " " | sed 's/ /°C /g' | sed 's/  $//'
diff --git b/polybar/.config/polybar/polybar-scripts/system-nvidia-smi.sh a/polybar/.config/polybar/polybar-scripts/system-nvidia-smi.sh
index 52bad3f..1348baf 100755
--- b/polybar/.config/polybar/polybar-scripts/system-nvidia-smi.sh
+++ a/polybar/.config/polybar/polybar-scripts/system-nvidia-smi.sh
@@ -1,3 +1,7 @@
 #!/bin/sh
 
-nvidia-smi --query-gpu=utilization.gpu --format=csv,noheader,nounits | awk '{ print "GPU",""$1"","%"}'
+# nvidia-smi --query-gpu=utilization.gpu --format=csv,noheader,nounits | awk '{ print "GPU",""$1"","%"}'
+
+# My system does not support the flag --query-gpu=utilization.gpu
+# That's why I modify the flag to show instead the used video memory.
+nvidia-smi --query-gpu=memory.used --format=csv,noheader,nounits | awk '{ print ""$1"","MiB"}'
diff --git b/polybar/.config/polybar/polybar-scripts/system-usb-udev.sh a/polybar/.config/polybar/polybar-scripts/system-usb-udev.sh
index ea62b68..64631d3 100755
--- b/polybar/.config/polybar/polybar-scripts/system-usb-udev.sh
+++ a/polybar/.config/polybar/polybar-scripts/system-usb-udev.sh
@@ -17,7 +17,7 @@ usb_print() {
         fi
         counter=$((counter + 1))
 
-        output="$output$space#1 $unmounted"
+        output="$output$space%{T4} %{T} $unmounted"
     done
 
     for mounted in $(echo "$devices" | jq -r '.blockdevices[] | select(.type == "part") | select(.rm == true) | select(.mountpoint != null) | .size'); do
@@ -28,7 +28,7 @@ usb_print() {
         fi
         counter=$((counter + 1))
 
-        output="$output$space#2 $mounted"
+        output="$output$space%{T4}%{T} $mounted"
     done
 
     echo "$output"
@@ -52,11 +52,16 @@ case "$1" in
         devices=$(lsblk -Jplno NAME,TYPE,RM,MOUNTPOINT)
 
         for mount in $(echo "$devices" | jq -r '.blockdevices[] | select(.type == "part") | select(.rm == true) | select(.mountpoint == null) | .name'); do
-            udisksctl mount --no-user-interaction -b "$mount"
+            # udisksctl mount --no-user-interaction -b "$mount"
 
             # mountpoint=$(udisksctl mount --no-user-interaction -b $mount)
-            # mountpoint=$(echo $mountpoint | cut -d " " -f 4- | tr -d ".")
+            # mountpoint=$(echo $mountpoint | cut -d " " -f 4 | tr -d ".")
             # terminal -e "bash -lc 'filemanager $mountpoint'"
+
+            mountpoint=$(udisksctl mount --no-user-interaction -b "$mount")
+            mountpoint=$(echo "$mountpoint" | cut -d " " -f 4 | tr -d ".")
+            # termite -e "bash -lc 'nautilus $mountpoint'" &
+            nautilus "$mountpoint"
         done
 
         usb_update
diff --git b/polybar/.config/polybar/polybar-scripts/updates-pacman-aurhelper.sh a/polybar/.config/polybar/polybar-scripts/updates-pacman-aurhelper.sh
index 5971118..fb1a6c2 100755
--- b/polybar/.config/polybar/polybar-scripts/updates-pacman-aurhelper.sh
+++ a/polybar/.config/polybar/polybar-scripts/updates-pacman-aurhelper.sh
@@ -5,7 +5,6 @@ if ! updates_arch=$(checkupdates 2> /dev/null | wc -l ); then
 fi
 
 if ! updates_aur=$(yay -Qum 2> /dev/null | wc -l); then
-# if ! updates_aur=$(paru -Qum 2> /dev/null | wc -l); then
 # if ! updates_aur=$(cower -u 2> /dev/null | wc -l); then
 # if ! updates_aur=$(trizen -Su --aur --quiet | wc -l); then
 # if ! updates_aur=$(pikaur -Qua 2> /dev/null | wc -l); then
@@ -13,10 +12,10 @@ if ! updates_aur=$(yay -Qum 2> /dev/null | wc -l); then
     updates_aur=0
 fi
 
-updates=$((updates_arch + updates_aur))
+updates=$(("$updates_arch" + "$updates_aur"))
 
 if [ "$updates" -gt 0 ]; then
-    echo "# $updates"
+    echo "  $updates"
 else
     echo ""
 fi
