diff -ruN --color ./polybar-scripts/info-hackspeed.sh ./polybar-scripts_custom/info-hackspeed.sh
--- ./polybar-scripts/info-hackspeed.sh	2021-01-30 16:50:16.408915278 +0100
+++ ./polybar-scripts_custom/info-hackspeed.sh	2021-01-30 16:50:14.668924826 +0100
@@ -1,12 +1,13 @@
 #!/bin/sh
 # shellcheck disable=SC2016,SC2059
 
-KEYBOARD_ID="AT Translated Set 2 keyboard"
+KEYBOARD_NAME="Logitech Wireless Keyboard PID:4023"
+KEYBOARD_ID=$(xinput list | grep "$KEYBOARD_NAME" | cut -d '=' -f 2 | awk '{ print $1 }' | tail -1)
 
 # cpm: characters per minute
 # wpm: words per minute (1 word = 5 characters)
 METRIC=cpm
-FORMAT="# %d $METRIC"
+FORMAT="%d $METRIC"
 
 INTERVAL=20
 
@@ -20,8 +21,7 @@
 	qwerty) CONDITION='($3 >= 10 && $3 <= 19) || ($3 >= 24 && $3 <= 33) || ($3 >= 37 && $3 <= 53) || ($3 >= 52 && $3 <= 58)'; ;;
 	azerty) CONDITION='($3 >= 10 && $3 <= 19) || ($3 >= 24 && $3 <= 33) || ($3 >= 37 && $3 <= 54) || ($3 >= 52 && $3 <= 57)'; ;;
 	qwertz) CONDITION='($3 >= 10 && $3 <= 20) || ($3 >= 24 && $3 <= 34) || ($3 == 36) || ($3 >= 38 && $3 <= 48) || ($3 >= 52 && $3 <= 58)'; ;;
-    dvorak) CONDITION='($3 >= 10 && $3 <= 19) || ($3 >= 27 && $3 <= 33) || ($3 >= 38 && $3 <= 47) || ($3 >= 53 && $3 <= 61)'; ;;
-    dontcare) CONDITION='1'; ;; # Just register all key presses, not only letters and numbers
+        dontcare) CONDITION='1'; ;; # Just register all key presses, not only letters and numbers
 	*) echo "Unsupported layout \"$LAYOUT\""; exit 1; ;;
 esac
 
diff -ruN --color ./polybar-scripts/info-softwarecounter.py ./polybar-scripts_custom/info-softwarecounter.py
--- ./polybar-scripts/info-softwarecounter.py	2021-01-30 16:50:15.025589536 +0100
+++ ./polybar-scripts_custom/info-softwarecounter.py	2021-01-30 16:50:14.668924826 +0100
@@ -14,36 +14,36 @@
 
 # program/process names and corresponding gylphs
 guis = OrderedDict({
-    'terminals': '#',
-    'chromes': '#',
-    'firefox': '#',
-    'skypeforlinux': '#',
-    'filemanager': '#',
-    'remote-desktop': '#',
-    'pdfviewer': '#',
-    'image': '#',
+    'terminals': '%{T10}%{T}',
+    'chromes': '%{T10}%{T}',
+    'firefox': '%{T10}%{T}',
+    'skypeforlinux': '%{T10}瑩%{T}',
+    'filemanager': '%{T10}%{T}',
+    'remote-desktop': '%{T10}ﲾ%{T}',
+    'pdfviewer': '%{T10}%{T}',
+    'image': '%{T10}%{T}',
 })
 
 processes = OrderedDict({
-    'vims': '#',
-    'ssh': '#',
-    'updater': '#',
+    'vims': '%{T10}%{T}',
+    'ssh': '%{T10}%{T}',
+    'updater': '%{T10}%{T}',
 })
 
 # combine counts of program/process names in the tuple
 # the resulting glpyh used will be that of the corresponding key
 combine_guis = {
-    'terminals': ('termite', 'terminator', 'urxvt'),
+    'terminals': ('xfce4-terminal', 'termite', 'terminator', 'urxvt'),
     'chromes': ('chromium', 'chrome'),
-    'filemanger': ('nemo', 'thunar', 'dolphin', 'nautilus', 'pcmanfm'),
+    'filemanger': ('nemo', 'thunar', 'dolphin', 'nautilus', 'pcmanfm', 'Archivos'),
     'remote-desktop': ('TeamViewer', ),
     'pdfviewer': ('evince', 'okular', 'zathura'),
     'image': ('gthumb', 'shotwell', 'deepin-image-vi'),
 }
 
 combine_proccesses = {
-    'vims': ('nvim', 'vim'),
-    'updater': ('pacman', 'yay', 'trizen', 'yaourt'),
+    'vims': ('nvim', 'vim', 'atom'),
+    'updater': ('pacman', 'yay', 'trizen', 'yaourt', 'makepkg', 'auracle'),
 }
 
 
diff -ruN --color ./polybar-scripts/openweathermap-fullfeatured.sh ./polybar-scripts_custom/openweathermap-fullfeatured.sh
--- ./polybar-scripts/openweathermap-fullfeatured.sh	2021-01-30 16:50:15.755585530 +0100
+++ ./polybar-scripts_custom/openweathermap-fullfeatured.sh	2021-01-30 16:50:14.668924826 +0100
@@ -3,24 +3,24 @@
 get_icon() {
     case $1 in
         # Icons for weather-icons
-        01d) icon="";;
-        01n) icon="";;
-        02d) icon="";;
-        02n) icon="";;
-        03*) icon="";;
-        04*) icon="";;
-        09d) icon="";;
-        09n) icon="";;
-        10d) icon="";;
-        10n) icon="";;
-        11d) icon="";;
-        11n) icon="";;
-        13d) icon="";;
-        13n) icon="";;
-        50d) icon="";;
-        50n) icon="";;
-        *) icon="";
-
+        01d) icon="%{T8} %{T}";;
+        01n) icon="%{T8} %{T}";;
+        02d) icon="%{T8} %{T}";;
+        02n) icon="%{T8} %{T}";;
+        03*) icon="%{T8} %{T}";;
+        04*) icon="%{T8} %{T}";;
+        09d) icon="%{T8} %{T}";;
+        09n) icon="%{T8} %{T}";;
+        10d) icon="%{T8} %{T}";;
+        10n) icon="%{T8} %{T}";;
+        11d) icon="%{T8} %{T}";;
+        11n) icon="%{T8} %{T}";;
+        13d) icon="%{T8} %{T}";;
+        13n) icon="%{T8} %{T}";;
+        50d) icon="%{T8} %{T}";;
+        50n) icon="%{T8} %{T}";;
+        *) icon="%{T8} %{T}";
+        
         # Icons for Font Awesome 5 Pro
         #01d) icon="";;
         #01n) icon="";;
@@ -38,7 +38,7 @@
         #*) icon="";
     esac
 
-    echo $icon
+    echo "$icon"
 }
 
 get_duration() {
@@ -52,8 +52,8 @@
 
 }
 
-KEY=""
-CITY=""
+KEY="a94a5e59327331833dd5e93d9b64885c"
+CITY="Getafe,ES"
 UNITS="metric"
 SYMBOL="°"
 
@@ -89,11 +89,11 @@
 
 
     if [ "$current_temp" -gt "$forecast_temp" ]; then
-        trend=""
+        trend="%{T8}%{T}"
     elif [ "$forecast_temp" -gt "$current_temp" ]; then
-        trend=""
+        trend="%{T8} %{T}"
     else
-        trend=""
+        trend="%{T8}%{T}"
     fi
 
 
@@ -102,11 +102,11 @@
     now=$(date +%s)
 
     if [ "$sun_rise" -gt "$now" ]; then
-        daytime=" $(get_duration "$((sun_rise-now))")"
+        daytime="%{T8}%{T} $(get_duration "$((sun_rise-now))")"
     elif [ "$sun_set" -gt "$now" ]; then
-        daytime=" $(get_duration "$((sun_set-now))")"
+        daytime="%{T8}%{T} $(get_duration "$((sun_set-now))")"
     else
-        daytime=" $(get_duration "$((sun_rise-now))")"
+        daytime="%{T8}%{T} $(get_duration "$((sun_rise-now))")"
     fi
 
     echo "$(get_icon "$current_icon") $current_temp$SYMBOL  $trend  $(get_icon "$forecast_icon") $forecast_temp$SYMBOL   $daytime"
diff -ruN --color ./polybar-scripts/system-cpu-temppercore.sh ./polybar-scripts_custom/system-cpu-temppercore.sh
--- ./polybar-scripts/system-cpu-temppercore.sh	2021-01-30 16:50:17.252243984 +0100
+++ ./polybar-scripts_custom/system-cpu-temppercore.sh	2021-01-30 16:50:14.668924826 +0100
@@ -1,3 +1,8 @@
 #!/bin/sh
 
-sensors | grep Core | awk '{print substr($3, 2, length($3)-5)}' | tr "\\n" " " | sed 's/ /°C  /g' | sed 's/  $//'
+#sensors | grep Core | awk '{print substr($3, 2, length($3)-5)}' | tr "\\n" " " | sed 's/ /°C  /g' | sed 's/  $//'
+
+# This is a more accurate way attending to my `sensors` output,
+# though I don't think these the core temperatures, I would say
+# these are another chipsets temps; but at least is something.
+sensors | grep "temp[0-9]" | awk '{print substr($2, 2, length($2)-3)}' | tr "\\n" " " | sed 's/ /°C /g' | sed 's/  $//'
diff -ruN --color ./polybar-scripts/system-nvidia-smi.sh ./polybar-scripts_custom/system-nvidia-smi.sh
--- ./polybar-scripts/system-nvidia-smi.sh	2021-01-30 16:50:17.732241350 +0100
+++ ./polybar-scripts_custom/system-nvidia-smi.sh	2021-01-30 16:50:14.668924826 +0100
@@ -1,3 +1,7 @@
 #!/bin/sh
 
-nvidia-smi --query-gpu=utilization.gpu --format=csv,noheader,nounits | awk '{ print "GPU",""$1"","%"}'
+# nvidia-smi --query-gpu=utilization.gpu --format=csv,noheader,nounits | awk '{ print "GPU",""$1"","%"}'
+
+# My system does not support the flag --query-gpu=utilization.gpu
+# That's why I modify the flag to show instead the used video memory.
+nvidia-smi --query-gpu=memory.used --format=csv,noheader,nounits | awk '{ print ""$1"","MiB"}'
diff -ruN --color ./polybar-scripts/system-usb-udev.sh ./polybar-scripts_custom/system-usb-udev.sh
--- ./polybar-scripts/system-usb-udev.sh	2021-01-30 16:50:18.358904578 +0100
+++ ./polybar-scripts_custom/system-usb-udev.sh	2021-01-30 16:50:14.668924826 +0100
@@ -17,7 +17,7 @@
         fi
         counter=$((counter + 1))
 
-        output="$output$space#1 $unmounted"
+        output="$output$space%{T4} %{T} $unmounted"
     done
 
     for mounted in $(echo "$devices" | jq -r '.blockdevices[] | select(.type == "part") | select(.rm == true) | select(.mountpoint != null) | .size'); do
@@ -28,7 +28,7 @@
         fi
         counter=$((counter + 1))
 
-        output="$output$space#2 $mounted"
+        output="$output$space%{T4}%{T} $mounted"
     done
 
     echo "$output"
@@ -60,7 +60,8 @@
 
             mountpoint=$(udisksctl mount --no-user-interaction -b "$mount")
             mountpoint=$(echo "$mountpoint" | cut -d " " -f 4 | tr -d ".")
-            termite -e "bash -lc 'mc $mountpoint'" &
+            # termite -e "bash -lc 'nautilus $mountpoint'" &
+            nautilus "$mountpoint"
         done
 
         usb_update
diff -ruN --color ./polybar-scripts/updates-pacman-aurhelper.sh ./polybar-scripts_custom/updates-pacman-aurhelper.sh
--- ./polybar-scripts/updates-pacman-aurhelper.sh	2021-01-30 16:50:18.798902163 +0100
+++ ./polybar-scripts_custom/updates-pacman-aurhelper.sh	2021-01-30 16:50:14.668924826 +0100
@@ -15,7 +15,7 @@
 updates=$(("$updates_arch" + "$updates_aur"))
 
 if [ "$updates" -gt 0 ]; then
-    echo "# $updates"
+    echo "  $updates"
 else
     echo ""
 fi
